web/src/ui_init.js

```diff
@@ -553,16 +555,25 @@ export function initialize_everything(state_data) {
             const sub = sub_store.get(stream_id);
             sidebar_ui.hide_all();
             popovers.hide_all();
-            message_view.show(
-                [
-                    {
-                        operator: "stream",
-                        operand: sub.stream_id.toString(),
-                    },
-                ],
-                {trigger},
-            );
+
+            const terms = [
+                {
+                    operator: "channel",
+                    operand: sub.stream_id.toString(),
+                },
+            ];
+            if (
+                user_settings.web_channel_default_view ===
+                web_channel_default_view_values.channel_feed.code
```

Comment:
Hmm, I think the "first topic" option should go through this code path too, right?

Reply:
No, since `on_stream_click` is only called when we don't want to navigate the the first topic in the channel.

See `set_event_handlers`:


```js
        if (
            // We expect `on_stream_click` to handle other values of `web_channel_default_view`.
            user_settings.web_channel_default_view !==
            web_channel_default_view_values.top_topic_in_channel.code
        ) {
            on_stream_click(stream_id, "sidebar");
            return;
        }

```

I will add a comment to clarify this.

Reply:
```
            // This function handles cases when we don't want user
            // to navigate to the first topic in channel on clicking
            // the channel name in the left sidebar.
```

Reply:
I changed the approach here since we using a different URL now. So, this code is no longer present.